app/views/documents/_document.html.erb

```diff
@@ -1,4 +1,9 @@
-<div class="uploaded_document">
-  <%= f.mb_input_field :document, nil,
-    options: { value: File.basename(document), disabled: true } %>
+<div class="doc-preview">
+  <div class="doc-preview__info">
+    <h4><%= File.basename(url || '{{filename}}') %></h4>
```

Comment:
where this `filename` is coming from feels kind of mysterious. Any way you can think of that would make it more obvious?